sos/report/plugins/ceph_osd.py

```diff
@@ -39,9 +39,62 @@ def setup(self):
 
         self.add_cmd_output([
             "ceph-disk list",
-            "ceph-volume lvm list",
+            "ceph-volume lvm list"
         ])
 
+        ceph_cmds = [
+           "bluestore bluefs available",
+           "config diff",
+           "config show",
+           "dump_blacklist",
+           "dump_blocked_ops",
+           "dump_historic_ops_by_duration",
+           "dump_historic_slow_ops",
+           "dump_mempools",
+           "dump_ops_in_flight",
+           "dump_op_pq_state",
+           "dump_osd_network",
+           "dump_reservations",
+           "dump_watchers",
+           "log dump",
+           "perf dump",
+           "perf histogram dump",
+           "objecter_requests",
+           "ops",
+           "status",
+           "version",
+        ]
+
+        osd_ids = []
+        out = self.exec_cmd('ps -u ceph -o args')
+
+        if out['status'] != 0:
+            return
```

Comment:
This will mean **whole remaining code in `setup()`** will be ignored and no command output from below commands will be collected - is that intentional?

Reply:
Agreed, we don't need to return here. If the command status is non-zero, then the following logic to populate osd_ids can be skipped, which will in turn make the `add_cmd_output()` calls a no-op. But, by not issuing a return here, we still hit the `add_forbidden_path()` call, without needing to move it around as well within the flow of `setup()`.

Reply:
done, thanks.